src/server/server_family.cc

```diff
@@ -299,6 +300,12 @@ bool ValidateServerTlsFlags() {
     has_auth = true;
   }
 
+  // Allow TLS without authentication for memcached protocol
+  // We check if memcached_port is enabled, as this is a static check during startup
+  if (GetFlag(FLAGS_memcached_port) > 0) {
```

Comment:
The problem with this is that it does not check the if the connection is `redis` or `memcache`. 

So your test would pass if it included:

```
    c = await aioredis.Redis(**with_ca_tls_client_args)
    res = await c.execute_command("set foo bar") 
    assert res == b"OK"
```

We did not set `requirepass` and `with_tls_server_args` does not use a `ca` so the command should have failed to authenticate yet it did. 

The solution is to check which protocol is used before you set `has_auth` to true

Reply:
ValidateServerTlsFlags is called during the server startup - there are no connections here. 
Server with memcache port won't require passwords for tls. That's what I asked .

Reply:
> Server with memcache port won't require passwords for tls.
The way I understood it was that you wanted to only apply this for the memcache port and the reason I raised this is
because we now allow launching dragonfly without `requirepass` or `ca_dir` and both `redis` and `memcache` connections can now connect without requirepass.

Makes sense now :)